hal/halx86/generic/up.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * FILE:        hal/halx86/generic/up.c
+ * PURPOSE:     Core source file for UP alternative functions
+ * PROGRAMMERS:  Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include <hal.h>
+#define NDEBUG
+#include <debug.h>
+
+/* FUNCTIONS *****************************************************************/
+
+VOID
+NTAPI
+HalRequestIpi(KAFFINITY TargetProcessors)
+{
+    /* This should never be called in UP mode */
+    __debugbreak();
```

Comment:
Do we really need this symbol ? If it's never called, it shouldn't even be there, right ?

Reply:
This and HalStartNextProcessor gets built with every single one of our Hals, Instead of having them in a processor.c thats associated with every different Hal, I threw these functions in this specific file so we easily build either the Uniprocessor or Multiprocessor count part.

This is the same functionality as before I just moved the functions into a better spot so we can actually build the CONFIG_SMP counterparts.

Reply:
I see they are in the exports of the hal
Unless we want to restore the preprocessed spec thingie (I personnally don't), I guess we have to live with that.

Wouldn't a bugcheck be what we want there ? We surely don't want a SMP kernel calling into a MP hal, do we ?